### PR TITLE
Documentation updates

### LINKED ISSUE #555: Requesting v0.7 documentation improvements/fixes
### Description

This is a list of things I found on the documentation of the new release that could be improved. Most are minor things.
1. YAML schema: description is missing tabs beyond the second level for some items, making it hard to read. Examples: `time_subset`, `parameters`, `tech_groups`.
2. Model configuration: it would be good to specify if the dates in `time_subset` are inclusive (does `['2005-01-01', '2005-01-05']` run up to or include May?)
3. Tab 2 or tab 4 usage is a bit inconsistent. There is a statement advocating for tab 2, but a good amount of examples use tab 4.
4. Specify checks for `techs`. For `cost_` parameters, the documentation mentions they _must_ define a cost class, but it's unclear what that means or what the code will check for. Is it an `index` or a `dims`? Will the library return an error if the user sets this incorrectly? `nodes` has something similar for `dims`: if the users specifies `nodes`, will the code catch it?
5. `nodes` deactivation: move the `nodes` deactivation comment from the `techs` section to the `nodes` section, to make it easier to find.
6. `nodes` using `constraints` to modify techs: in the nodes section it's mentioned that you overwrite `techs` settings by using `constraints`. I think this term is a bit confusing (it's not used anywhere else before, and it's seldom mentioned in the documentation). Is it really necessary? Otherwise, I'd just remove it since it's clear that anything below a `techs` will be a parameter, and `costs_` already have standardised naming.
7. Tabular data: I recommend showing the tabular data as tables instead of comma text, to make it easier to read.
8. Inbuilt math vs custom-math: I think the activation of Inter-cluster storage via custom math is a bit confusing and hard to find. Maybe moving it to a "pre-defined custom math" section would solve this. As it stands, the inbuilt math section mostly talks about custom math.
9. Math readability: most equations are quite hard to follow because they get "cut" due to the long names. Defining the conditions above the constraints would help with this:

> Initial conditions (for all timestep in timesteps... etc):
>- Equation_1... if condition_1...
>- Equation_2... if condition_2...

Or (for an even more readable alternative):
> Initial conditions:
> - if condition_1:
>   Equation_1

10. `foreach`, `where` math syntax examples: when reading the section, it's a bit difficult to see how these are used since there is no complete example. Adding one at the start would solve this. Or moving the `equation` section to the top, since it's where most of these terms are used.
11. Math components and Math syntax: the custom math section says to read syntax, then components, but the order and the way the text is written seems to suggest the opposite.

### Related links

1. https://calliope.readthedocs.io/en/v0.7.0.dev2/reference/config_schema/#model-configuration-schema
2. https://calliope.readthedocs.io/en/v0.7.0.dev2/creating/config/
3. https://calliope.readthedocs.io/en/v0.7.0.dev2/creating/techs/ (example of tab 4 usage)
4. https://calliope.readthedocs.io/en/v0.7.0.dev2/creating/techs/ (unclear if there are checks for `cost_`)
5. https://calliope.readthedocs.io/en/v0.7.0.dev2/creating/techs/ (`nodes` comment that is hard to find)
6. https://calliope.readthedocs.io/en/v0.7.0.dev2/creating/nodes/ (`constraints` to modify `techs`)
7. https://calliope.readthedocs.io/en/v0.7.0.dev2/creating/data_sources/#__tabbed_3_1 (showing tabular data as tables)
8. https://calliope.readthedocs.io/en/v0.7.0.dev2/math/ (inbuilt math vs custom math).
9. https://calliope.readthedocs.io/en/v0.7.0.dev2/math/base/ (making math more readable).
10. https://calliope.readthedocs.io/en/v0.7.0.dev2/custom_math/syntax/ (`where`, `foreach` usage is unclear)
11. https://calliope.readthedocs.io/en/v0.7.0.dev2/custom_math/ (reverse suggestion to read Math syntax then components)

### Version

v0.7.0

### Proposed change

Some suggested updates to the documentation.

### PR DESCRIPTION
**Pull request by brynpickering**
Fixes #555 

## Changes made in this Pull Request

- rename "custom math" to "user-defined math" across the board.
- rename "inbuilt math" to "pre-defined math" across the board.
- 4-space -> 2-space indentation in all YAML snippets.
- Added graphic to help visualise how the math syntax gets converted to a matrix operation.
- Cleaned up some confusing text that wasn't correctly updated from old docs.
- Update latex math representation to improve readability (as per @irm-codebase's suggestion).

Reviewer checklist:

- [ ] Test(s) added to cover contribution
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Coverage maintained or improved

### CODE REVIEW

**Comment by brynpickering** [comment]
@irm-codebase and @sjpfenninger, ready for review. 

@irm-codebase, you can view the documentation with the PR changes by selecting "Details" of the readthedocs CI check (once it has a green tick mark next to it):

image

**Comment by irm-codebase** [comment]
Most things look great so far!
I only found one funky detail in the schema:
image

File: src/calliope/backend/latex_backend_model.py
Comment (irm-codebase) [inline src/calliope/backend/latex_backend_model.py:1]: One little comment here:
There's a comma that is acting funny at the end of the $\forall$ line:
image

Not the most important thing, though.

**Comment by irm-codebase** [comment]
> > Most things look great so far! I only found one funky detail in the schema: image
> 
> Yeah, this is an issue with the upstream dependency. It can't handle "OneOf"/"AnyOf"/"AllOf" schema elements. Another reason perhaps to have a different place to describe parameters.

In that case I'd recommend to add parameters to the math section (as a table, maybe?). To be honest, I am struggling a bit when translating an older model (the Italy one) to test the pathways because of this, since not all cases are listed.

**Review by irm-codebase — DISMISSED** [review]
I think all the changes are good so far!
I will open an issue to add predefined parameters to the documentation.
